save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/RegistrationView.kt

```diff
@@ -0,0 +1,276 @@
+/**
+ * A view for registration
+ */
+
+@file:Suppress("FILE_WILDCARD_IMPORTS", "WildcardImport")
+
+package com.saveourtool.save.frontend.components.views
+
+import com.saveourtool.save.domain.ImageInfo
+
+import com.saveourtool.save.frontend.components.inputform.InputTypes
+import com.saveourtool.save.frontend.components.inputform.inputTextFormRequired
+import com.saveourtool.save.frontend.http.getUser
+import com.saveourtool.save.frontend.utils.*
+import com.saveourtool.save.frontend.utils.classLoadingHandler
+import com.saveourtool.save.info.UserInfo
+import com.saveourtool.save.utils.AvatarType
+import com.saveourtool.save.v1
+import com.saveourtool.save.validation.FrontendRoutes
+import com.saveourtool.save.validation.isValidName
+
+import csstype.ClassName
+import org.w3c.dom.HTMLInputElement
+import org.w3c.dom.asList
+import org.w3c.fetch.Headers
+import org.w3c.xhr.FormData
+import react.*
+import react.dom.aria.ariaLabel
+import react.dom.events.ChangeEvent
+import react.dom.html.ButtonType
+import react.dom.html.InputType
+import react.dom.html.ReactHTML.button
+import react.dom.html.ReactHTML.div
+import react.dom.html.ReactHTML.form
+import react.dom.html.ReactHTML.h1
+import react.dom.html.ReactHTML.img
+import react.dom.html.ReactHTML.input
+import react.dom.html.ReactHTML.label
+import react.dom.html.ReactHTML.main
+import react.dom.html.ReactHTML.span
+
+import kotlinx.browser.window
+import kotlinx.coroutines.launch
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+
+/**
+ * `Props` retrieved from router
+ */
+external interface RegistrationProps : PropsWithChildren {
+    /**
+     * Currently logged in user or null
+     */
+    var userName: String?
```

Comment:
Can we pass `userInfo` from `App.kt` instead of just name?

Reply:
Ok. Fixed